src/app/index.ts

```diff
@@ -97,27 +108,38 @@ module.exports = class TheiaExtension extends Base {
         this.sourceRoot(__dirname + '/../../templates');
     }
 
-    prompting() {
-        if (!(this.options as any).extensionName)
-            return this.prompt([{
-                type: 'input',
-                name: 'name',
-                message: "The extension's name",
-                default: path.parse(process.cwd()).name
-            },
-            {
+    async prompting() {
+        let extensionType = (this.options as any).extensionType;
+        const inExtensionType = Object.values(ExtensionType).includes(extensionType);
```

Comment:
This doesn't compile with the current `tsconfig` settings. Please change to `(<any>Object).` instead of `Object.`